ext/session/session.c

```diff
@@ -2227,14 +2356,23 @@ static PHP_FUNCTION(session_destroy)
 		return;
 	}
 
+	if (PS(session_status) != php_session_active) {
+		php_error_docref(NULL, E_WARNING, "Trying to destroy uninitialized session");
```

Comment:
I'm not sure about this one... There might be code that calls destroy repeatedly, just in case - e.g. error handlers, etc. - and putting warning there does not help much. Since the session stays destroyed, there's not real reason to react with a failure error. I'd just silently return, maybe return false at worst. Warning should be when whatever you tried to do failed, but here it didn't actually fail - it was just unnecessary.